azurerm/utils/base64.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import "encoding/base64"
+
+// base64Encode encodes data if the input isn't already encoded using
+// base64.StdEncoding.EncodeToString. If the input is already base64 encoded,
+// return the original input unchanged.
+func Base64EncodeIfNot(data string) string {
+	// Check whether the data is already Base64 encoded; don't double-encode
+	if Base64IsEncoded(data) {
+		return data
+	}
+	// data has not been encoded encode and return
+	return base64.StdEncoding.EncodeToString([]byte(data))
+}
+
+func Base64IsEncoded(data string) bool {
```

Comment:
since this isn't used outside of this package we can make this private:

```suggestion
func base64IsEncoded(data string) bool {
```